IPs/RSKIP170.md

```diff
@@ -0,0 +1,71 @@
+# Peg-in to any address
+
+|RSKIP          |170           |
+| :------------ |:-------------|
+|**Title**      |Peg-in to any address |
+|**Created**    |01-SEP-20 |
+|**Author**     |MI |
+|**Purpose**    |USa |
+|**Layer**      |Core |
+|**Complexity** |2 |
+|**Status**     |Draft |
+
+## Abstract
+
+This RSKIP proposes a way for users to be able to lock funds from BTC, indicating the RSK address in which they want their RBTCs transferred to and a BTC refund address in case the peg-in can not be completed for some reason.
+
+## Motivation
+
+Allowing users to lock funds to any address in RSK.
+
+## Specification
+
+The BTC transaction sent to the federation must contain at most one output with value 0 and OP_RETURN op code followed by the following data:
+- Protocol version (2 bytes). Currently **1**
+- A valid RSK address (20 bytes)
+- [Optional] A valid BTC refund address in the following format:
+    - Address type (1 byte)
+        - 1 for P2PKH
+        - 2 for P2SH
+    - Hash needed to build refund address (20 bytes)
+        - For P2PKH type address: public key hash 
+        - For P2SH type address: script hash
+
+### Example output
+```
+{
+      "value": 0.00000000,
+      "n": 1,
+      "scriptPubKey": {
+        "asm": "OP_RETURN 00010e537aad84447a2c2a7590d5f2665ef5cf9b667a014f4c767a2d308eebb3f0f1247f9163c896e0b7d2",
+        "hex": "6a2b00010e537aad84447a2c2a7590d5f2665ef5cf9b667a014f4c767a2d308eebb3f0f1247f9163c896e0b7d2",
+        "type": "nulldata"
+      }
+    }
+```
+Payload included: `00010e537aad84447a2c2a7590d5f2665ef5cf9b667a014f4c767a2d308eebb3f0f1247f9163c896e0b7d2`
+
+- First 2 bytes correspond to the version number: `0001`
+- Next 20 bytes indicate rsk destination address: `0e537aad84447a2c2a7590d5f2665ef5cf9b667a`
+- Next 1 byte indicating btc refund address type, in this case P2PKH: `01`
+- Finally 20 bytes indicating the public key hash that will be used to get the btc refund address: `4f4c767a2d308eebb3f0f1247f9163c896e0b7d2`
+
+### Notes
+- Only one output with OP_RETURN op code is allowed in the pegin transaction. Transactions with more than one OP_RETURN outputs will be rejected.
```

Comment:
I think that rejecting in case there are more than one OP_RETURN is a bad idea. This is because there are use cases related with Mining products (i.e. decentralized pools) that we want to send bitcoins directly from coinbase transactions to a smart contract. At the same time other blockchains require OP_RETURNs to merge-mine with Bitcoin, so it won't be possible to do both at the same time, losing revenue.

Therefore I suggest that before protocol version, we add a 4-byte magic number (.i.e RSKT) for RSK Transfer. Then only forbid more than one OP_RETURN  with the RSKT magic world.

Reply:
Added specification that `RSKT` in ascii and hexa encoded (`52534b54`) should be added as prefix to the payload in OP_RETURN output